DMPRoadmap/roadmap
# Fix bug when editing question on Template

Changes proposed in this PR:
- Add a spec to test behaviour for editing question
- Clean up some smelly methods
- Ensure annotations aren't created unless they have text
- Clean up the templates controller index actions to avoid mis-using `render`
- Remove bang! variants of save methods used in controllers. When these are used in an `if` statement, we don't want to raise exceptions.
